hbase-it/pom.xml

```diff
@@ -154,10 +154,29 @@
       <type>test-jar</type>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.apache.hbase</groupId>
+      <artifactId>hbase-server</artifactId>
+      <type>test-jar</type>
```

Comment:
Why are there `compile` scope dependencies on test jars? Is there a plan to move the required code into a `src/main/java`?

Reply:
yes I am moving these classes from src/test/java -> src/main/java as discussed with Sean and Andrew that any code under test with an audience and stability annotation is not meaningful and anti-pattern.

Reply:
Yes some classes in `hbase-it/src/main/java` have dependencies on classes present in `hbase-server/src/test/java` for ex: some `DistributedHBaseCluster` extends `HBaseCluster` which is present in `hbase-server/src/test/java`.

Reply:
@ndimiduk Classes which we are moving are using many test classes of other modules, 
we can move those classes to `src/main/java` but they have dependencies on many other test classes. So, to move these classes if we follow this route, how can we proceed without test dependency!?
CC: @apurtell @busbey

Reply:
This dependency graph goes deep, and makes this a non-trivial challenge to unravel. I agree we should be shipping a solid suite of tools for testing HBase and HBase applications. Sadly changing only the leaves of that graph is not sufficient. IMHO, you have to start at the root classes and work out to the leaves. I don't know if that effort is worth it when compared to building a new testing harness from scratch that's designed from the outset to be user-facing.

Reply:
@ndimiduk I moved the whole hierarchy in src/ on which chaos framework had a dependency, mostly classes are parent classes of `IntegrationTestingUtility`  and  `DistributedHBaseCluster` and their dependencies. Please take a look if this is the right way!!
Thanks.